code/controllers/stonedmc/subsystem/job.dm

```diff
@@ -97,6 +97,95 @@ SUBSYSTEM_DEF(job)
 	JobDebug("AR has failed, Player: [player], Mind: [player?.mind], Rank: [rank]")
 	return FALSE
 
+/proc/handle_squad(mob/M, rank, latejoin = FALSE)
+	var/strict = FALSE
+	var/datum/squad/P = SSjob.squads[M.client.prefs.preferred_squad]
+	var/datum/squad/R = SSjob.squads[pick(SSjob.squads)]
```

Comment:
We might want to add further checks for how squads are picked, but this is a good base.

Reply:
Yeah it's not very modular since it has hardcoded roles, just something I hacked together to make the job subsystem work plus preserve the "prefer squad over role" idea.